skyportal/models.py

```diff
@@ -1228,9 +1235,21 @@ class Candidate(Base):
 
 
 Source = join_model("sources", Group, Obj)
-Source.create = (
-    Source.read
-) = Source.update = Source.delete = accessible_by_group_members
+
+
+def source_create_access_logic(cls, user_or_token):
+    user_id = UserAccessControl.user_id_from_user_or_token(user_or_token)
+    query = DBSession().query(cls)
+    if not user_or_token.is_system_admin:
+        query = query.join(Group).join(GroupUser)
+        query = query.filter(GroupUser.user_id == user_id, GroupUser.can_save.is_(True))
+    return query
+
+
+Source.create = accessible_by_group_members & CustomUserAccessControl(
+    source_create_access_logic
+)
```

Comment:
```suggestion
Source.create = CustomUserAccessControl(
    source_create_access_logic
)
```
I don't think we need the `accessible_by_group_members` anymore as that will also be accounted for by the `query = query.filter(GroupUser.user_id == user_id,` part in the `source_create_access_logic`